src/model/event.rs

```diff
@@ -948,30 +947,47 @@ pub enum GatewayEvent {
 // Manual impl needed to emulate integer enum tags
 impl<'de> Deserialize<'de> for GatewayEvent {
     fn deserialize<D: Deserializer<'de>>(deserializer: D) -> StdResult<Self, D::Error> {
-        let mut map = JsonMap::deserialize(deserializer)?;
-
-        Ok(match remove_from_map(&mut map, "op")? {
+        #[derive(Debug, Clone, Deserialize)]
+        struct GatewayEventRaw<'a> {
+            op: Opcode,
+            #[serde(rename = "s")]
+            seq: Option<u64>,
+            #[serde(rename = "d")]
+            data: &'a RawValue,
+            #[serde(rename = "t")]
+            ty: Option<&'a str>,
+        }
+
+        let raw: GatewayEventRaw<'_> = Deserialize::deserialize(deserializer)?;
```

Comment:
```suggestion
        let raw = GatewayEventRaw::deserialize(deserializer)?;
```
More readable/simpler.

Reply:
fixed